virtualizarr/xarray.py

```diff
@@ -41,49 +42,84 @@ def open_virtual_dataset(
         If not provided will attempt to automatically infer the correct filetype from the the filepath's extension.
     drop_variables: list[str], default is None
         Variables in the file to drop before returning.
+    load_variables: list[str], default is None
+        Variables in the file to open using xarray's normal lazy indexing classes (i.e. to be accessible as numpy/dask arrays) instead of opening as instances of virtual_array_class.
+        Default is to open all variables as virtual arrays (i.e. ManifestArray).
     indexes : Mapping[str, Index], default is None
         Indexes to use on the returned xarray Dataset.
         Default is None, which will read any 1D coordinate data to create in-memory Pandas indexes.
         To avoid creating any indexes, pass indexes={}.
     virtual_array_class
         Virtual array class to use to represent the references to the chunks in each on-disk array.
         Currently can only be ManifestArray, but once VirtualZarrArray is implemented the default should be changed to that.
+
+    Returns
+    -------
+    vds
+        An xarray Dataset containing instances of virtual_array_cls for each variable, or normal lazily indexed arrays for each variable in load_variables.
     """
 
+    if drop_variables is None:
+        drop_variables = []
+    if load_variables is None:
+        load_variables = []
+
     # this is the only place we actually always need to use kerchunk directly
+    # TODO drop_variables for variables which will be loaded using xarray later anyway?
     vds_refs = kerchunk.read_kerchunk_references_from_file(
         filepath=filepath,
         filetype=filetype,
     )
 
-    if indexes is None:
-        # add default indexes by reading data from file
+    if indexes is None or load_variables != []:
         # TODO we are reading a bunch of stuff we know we won't need here, e.g. all of the data variables...
         # TODO it would also be nice if we could somehow consolidate this with the reading of the kerchunk references
-        ds = xr.open_dataset(filepath)
+        # TODO really we probably want a dedicated xarray backend that iterates over all variables only once
+        ds = xr.open_dataset(filepath, drop_variables=drop_variables)
```

Comment:
At some point it might be nice to have a dedicated function / xarray backend that only opens the legacy file once, instead of opening it once with kerchunk and once with with xarray.